tooling/templatize/cmd/entrypoint/cleanup/resourcedeleter.go

```diff
@@ -0,0 +1,840 @@
+		"Microsoft.Network/networkSecurityGroups",
+		"Microsoft.Insights/dataCollectionRules",
+		"Microsoft.Insights/dataCollectionEndpoints",
+		"Microsoft.ContainerInstance/containerGroups",
+	}
+
+	var resources []*armresources.GenericResourceExpanded
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to list resources: %w", err)
+		}
+		for _, resource := range page.Value {
+			if resource.ID != nil && resource.Type != nil && resource.Name != nil {
+				// Skip excluded types
+				excluded := false
+				for _, excludedType := range excludedTypes {
+					if strings.EqualFold(*resource.Type, excludedType) {
+						excluded = true
+						break
+					}
+				}
+				if !excluded {
+					resources = append(resources, resource)
+				}
+			}
+		}
+	}
+
+	if len(resources) == 0 {
+		d.logger.Info("No application resources found to delete")
+		return nil
+	}
+
+	d.logger.Info("Found application resources to delete", "count", len(resources))
+
+	// Delete each resource
+	stats := &deletionStats{}
+
+	for _, resource := range resources {
+		resourceName := *resource.Name
+		resourceType := *resource.Type
+		resourceID := *resource.ID
+
+		// Check for locks
+		if d.hasLocks(ctx, resourceID) {
+			d.logger.Info("Skipping locked resource", "name", resourceName, "type", resourceType)
+			stats.skipped++
+			continue
+		}
+
+		if d.dryRun {
+			d.logger.Info("[DRY RUN] Would delete resource", "name", resourceName, "type", resourceType)
+			stats.deleted++
+		} else {
+			if err := d.deleteResourceWithRetries(ctx, client, resourceID, resourceName, resourceType, 1); err != nil {
+				d.logger.Error(err, "Failed to delete resource", "name", resourceName, "type", resourceType)
+				stats.failed++
+				// Continue with other resources
+			} else {
+				stats.deleted++
+			}
+		}
+	}
+
+	d.logger.Info("Application resources deletion summary",
+		"deleted", stats.deleted,
+		"skipped", stats.skipped,
+		"failed", stats.failed)
+
+	return nil
+}
+
+// deleteResourceWithRetries attempts to delete a resource with retries
+func (d *resourceGroupDeleter) deleteResourceWithRetries(ctx context.Context, client *armresources.Client, resourceID, resourceName, resourceType string, maxRetries int) error {
+	var lastErr error
+
+	// Extract API version from resource type
+	// Azure SDK requires the API version to delete resources
+	apiVersion := d.getAPIVersionForResourceType(resourceType)
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		if attempt > 1 {
+			d.logger.Info("Retrying resource deletion", "name", resourceName, "type", resourceType, "attempt", attempt, "maxRetries", maxRetries)
+			time.Sleep(10 * time.Second)
+		}
+
+		// Begin delete with API version
+		poller, err := client.BeginDeleteByID(ctx, resourceID, apiVersion, nil)
+		if err != nil {
+			lastErr = err
+			continue
+		}
+
+		// Wait for completion
+		_, err = poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
```

Comment:
you probably want to do these in parallel - delete all of one type in parallel, wait on them all befor emoving on to a new type. this will be slow